lib/registries/addon/overview/route.ts

```diff
@@ -76,20 +81,40 @@ export default class Overview extends GuidRoute {
                 ),
                 institution: (institutions as SparseModel[]).map(institution => institution.name as string),
             };
-            const headTags = [...this.metaTags.getHeadTags(metaTagsData)];
-            if (provider && provider.assets && provider.assets.favicon) {
-                headTags.push({
-                    type: 'link',
-                    attrs: {
-                        rel: 'icon',
-                        href: provider.assets.favicon,
-                    },
-                });
+
+            let jsonLD: object = {};
+            let scriptTag: HeadTagDef[] = [];
+            try {
+                jsonLD = await this.scriptTags.returnStructuredData(id);
+                const jsonString: string = Object.entries(jsonLD) ?
+                    JSON.stringify(jsonLD) : JSON.stringify({ isAccessibleForFree : true });
+                const scriptTagData = {
+                    type: 'application/ld+json',
+                    content: jsonString,
+                };
+                scriptTag = await this.scriptTags.getHeadTags(scriptTagData);
+            } catch (e) {
+                const errorMessage = this.intl.t('general.structured_data.json_ld_retrieval_error');
+                captureException(e, { errorMessage });
+            }
```

Comment:
The reason tests were failing previously was because an `Error` was thrown in `returnStructuredData` because there were no mirage endpoint for the ajax call. And since the error is not caught, the `Uncaught Error` message was shown in the GA logs. If you have already handled the exception in `returnStructuredData`, then there is no error to catch here. In fact, the exception this try-catch block catches will most likely be the exception raised by `Object.entries(jsonLD)` or `this.scripTags.getHeadTags(scriptTagData)`, which are not `json_ld_retrieval_error`.

It is your choice where to catch and handle the exception. It could be here or it could be within `returnStructuredData`, but not both. It should also be noted that when we `throw` an exception, there must be some upper-level error handler to handle it, otherwise it would be an uncaught exception, which is bad. Another thing to remember about try-catch blocks is that we should only have whatever code that would throw exceptions within the `try` block, and only handle the respective exception in the `catch` block. When an exception is thrown within the `try` block, it would immediately jump to the `catch` block to handle the exception, and then return to execute whatever code that is after the `catch` block, unless there are some new exceptions thrown in the `catch` block, in which case execution would end because of uncaught exception error.